File: get_data.py
import requests 
import os
import argparse
import zipfile

parser = argparse.ArgumentParser()

def folder(save_dir):
    # Create target Directory if don't exist
    if not os.path.exists(save_dir):
        os.mkdir(save_dir)
        print(f"Directory {save_dir} Created ")
    
    return save_dir


def download_and_process_data(url, save_dir):
    # Check if the folder exisits, if not create

    filename = url.split('/')[-1]

    r = requests.get(url) # create HTTP response object 

    # Download the file
    print('save_dir: ', save_dir)
    print('filename: ', filename)
    filepath = os.path.join(save_dir, filename)
    print("Downloading the content")
    with open(filepath,'wb') as f: 
        f.write(r.content) 

    # Unzip the file if needed:
    if zipfile.is_zipfile(filepath):
        print("Unzipping the downloaded file")
        with zipfile.ZipFile(filepath, 'r') as zip_ref:
            zip_ref.extractall(save_dir)


if __name__ == '__main__':
    parser.add_argument('--url', type=str,
            required=False, help='The URL from where the data can be download',
            default='https://archive.ics.uci.edu/ml/machine-learning-databases/00222/bank.zip')

    parser.add_argument('--save_dir', type=folder,
            required=False, help='The folder where the data will be saved',
            default='./data')

    args = parser.parse_args()

    download_and_process_data(args.url, args.save_dir)

File: src/get_data.R
'Download data from given URL and extract in given folder if .zip file
Usage:
    get_data.R [--url=<url> --out_dir=<out_dir>]
    
Options:
    -h --help  Show this screen.
    --url=<url>  URL of file [default: https://archive.ics.uci.edu/ml/machine-learning-databases/00222/bank.zip]
    --out_dir=<out_dir>  Directory to save the data [default: ./data/raw/]
' -> doc

library(docopt)
library(tools)


arguments <- docopt(doc)

main <- function () {
	
	dir.create(arguments$out_dir, showWarnings = FALSE)

	filename <- gsub("^.*/", "", arguments$url)
	filepath <- file.path(arguments$out_dir, filename)
	if(!download_data(arguments$url, arguments$out_dir, filepath)){
		return()
	}
	
	
	extension <- file_ext(filename)
	if(extension == 'zip'){
		print('Unzipping the downloaded file')
		unzip_data(filepath, arguments$out_dir)
	} else {
		print(paste('Not unzipping. The file extension is not zip. It is ', extension))
	}
}

download_data <- function(url, dir, filepath) {
	return(tryCatch({
		dir.create(dir, recursive = TRUE)
		print('Downloading file')
		download.file(url, filepath)

		if (file.exists(filepath)){
			print(paste('File is stored in :', filepath))
			return(TRUE)
		} else {
			print('File has not been downloaded for some reason. Please check your internet connection')
			return(FALSE)
		}
	}, warning=function(cond) {
            	message(paste("URL caused a warning:", url))
		message('Check the URL!!')
            	return(FALSE)
        }, error = function(cond) {
		message(paste("URL does not seem to exist:", url))
            	return(FALSE)
	}))
}

unzip_data <- function(filepath, out_dir) {
	print(paste('Unzipping file: ', filepath))
	return(tryCatch({
		unzip(filepath, exdir=out_dir)
		return(TRUE)
	}, warning=function(cond) {
		message('Cannot unzip: Check file!!')
            	return(FALSE)
        }, error = function(cond) {
		message(paste("Cannot unzip: Check file!"))
            	return(FALSE)
	})) 
}

if (sys.nframe() == 0){
  main()
}
